### PR TITLE
Generate Express apps.

### PR DESCRIPTION
**Pull request by ddunkin**

### CODE REVIEW

**Comment by ddunkin** [comment]
@ejball this is a work in progress, but is it the right direction?

**Comment by ejball** [comment]
@ddunkin, this is awesome! Haven't read carefully but definitely seems like the right direction.

File: src/Facility.CodeGen.JavaScript/JavaScriptGenerator.cs [hunk 1]
```diff
@@ -255,6 +288,159 @@ protected override CodeGenOutput GenerateOutputCore(ServiceInfo service)
...
+								code.WriteLine($"return service.{methodName}(request)");
+
+								using (code.Indent())
+								{
+									using (code.Block(".then(result => {", "})"))
+									{
+										using (code.Block("if (result.error) {", "}"))
+										{
+											code.WriteLine("const status = result.error.code && standardErrorCodes[result.error.code] || 500;");
+											code.WriteLine("res.status(status).send(result.error.details);");
```
Comment (ejball) [inline src/Facility.CodeGen.JavaScript/JavaScriptGenerator.cs]: I'm pretty sure this should be `.send(result.error)`.

File: src/Facility.CodeGen.JavaScript/JavaScriptGenerator.cs [hunk 2]
```diff
@@ -316,24 +502,52 @@ private string RenderUriComponent(ServiceFieldInfo field, ServiceInfo service)
...
+			var fieldName = field.Name;
+
+			switch (fieldTypeKind)
+			{
+				case ServiceTypeKind.Enum:
+				case ServiceTypeKind.String:
+				case ServiceTypeKind.Bytes:
+					return value;
+				case ServiceTypeKind.Boolean:
+					return $"{value} === 'true'";
```
Comment (ejball) [inline src/Facility.CodeGen.JavaScript/JavaScriptGenerator.cs]: I really need to document the syntax for query string parameter values.

Consider returning null or undefined if the value isn't true or false. That's what the C# implementation currently does, though I wonder if it should return an invalid request.

Also, C# uses a case-insensitive match. But I think I will allow implementations to be case-sensitive. Certainly JS/TS clients would be a lot harder to write if case-insensitivity were allowed everywhere.

**Pull request merged**